language/move-prover/src/prelude.bpl

```diff
@@ -133,10 +133,15 @@ axiom v#ValueArray(EmptyValueArray) == MapConstValue(DefaultValue);
 function {:inline} AddValueArray(a: ValueArray, v: Value): ValueArray {
     ValueArray(v#ValueArray(a)[l#ValueArray(a) := v], l#ValueArray(a) + 1)
 }
-
 function {:inline} RemoveValueArray(a: ValueArray): ValueArray {
     ValueArray(v#ValueArray(a)[l#ValueArray(a) - 1 := DefaultValue], l#ValueArray(a) - 1)
 }
+function {:inline} RemoveIndexValueArray(a: ValueArray, i: int): ValueArray {
+    if i < 0 || i >= l#ValueArray(a) then a
```

Comment:
Why is this check needed? This function should never be called with index out of range. Rather, a procedure calling this should make a range check and abort before calling this.

Are you doing this because its called from a spec expression with index out of range? IIRC, this shouldn't be called from spec expressions.

I'm mostly asking because we should be careful in not adding redundant semantics to the prelude, because this can obfuscate what we are doing.

Reply:
OK. I will remove this check. As you said, this check is unnecessary because the procedures that call this function will/should never violate it.

Reply:
Done.

Reply:
I have a response to the next comment about using functions in specifications.  However, are these functions going to appear in specifications?